ci/azure-pipelines-dev.yml

```diff
@@ -169,80 +169,18 @@ stages:
           pip install --requirement reqs/requirements-dev-test.txt
           invoke docstrings
         name: DocstringLinter
-
     - job: unused_import_checker
       steps:
       - script: |
           pip install $(grep -E '^(ruff)' reqs/requirements-dev-contrib.txt)
           # https://www.flake8rules.com/rules/F401.html
           ruff --select F401 great_expectations tests
         name: UnusedImportChecker
-
     - job: check_repo_root_size
       steps:
         - script: ./ci/checks/check_repo_root_size.sh
           name: CheckRepoRootSize
 
-    - job: docs_snippet_checker
-      condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.DocsChanged'], true)
-      steps:
-        - script: |
-            yarn install
-          name: YarnInstall
-        - task: PythonScript@0
-          inputs:
-            scriptSource: 'filePath'
-            scriptPath: 'ci/checks/validate_docs_snippets.py'
-            failOnStderr: false
-          name: DocsSnippetChecker
-
-    - job: line_number_snippet_checker
-      condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.DocsChanged'], true)
-      steps:
-        - task: PythonScript@0
-          inputs:
-            scriptSource: 'filePath'
-            scriptPath: 'ci/checks/check_no_line_number_snippets.py'
-            failOnStderr: true
-          name: LineNumberSnippetChecker
-
-
-    - job: name_tag_snippet_checker
-      condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.DocsChanged'], true)
-      steps:
-        - task: PythonScript@0
-          inputs:
-            scriptSource: 'filePath'
-            scriptPath: 'ci/checks/check_only_name_tag_snippets.py'
-            failOnStderr: true
-          name: NameTagSnippetChecker
-
-    - job: integration_test_gets_run_checker
-      condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.DocsChanged'], true)
-      steps:
-        - task: PythonScript@0
-          inputs:
-            scriptSource: 'filePath'
-            scriptPath: 'ci/checks/check_integration_test_gets_run.py'
-            failOnStderr: true
-          name: IntegrationTestGetsRunChecker
-
-    - job: name_tag_snippet_referenced_checker
-      condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.DocsChanged'], true)
-      steps:
-        - task: PythonScript@0
-          inputs:
-            scriptSource: 'filePath'
-            scriptPath: 'ci/checks/check_name_tag_snippets_referenced.py'
-            failOnStderr: true
-          name: NameTagSnippetReferencedChecker
-    - job: public_api_report
-      steps:
-        - script: |
-            pip install --requirement reqs/requirements-dev-test.txt
-            invoke public-api
-          name: PublicAPIReport
```

Comment:
I think this `public_api_report` job is running in the `docs` pipeline but only if docs change. I think we also need to run it if GX changes. Agree on only running it in one pipeline (`docs`) but that pipeline should be modified to run this on any GX code change (e.g. adding `@public_api` decorator)